app.vue

```diff
@@ -1,7 +1,7 @@
 <template>
-  <LayoutHeader />
-  <LandingHeader />
-  <OverviewImage />
+  <LayoutHeader :on-click="onClick" />
+  <LandingHeader :on-click="onClick" />
+  <OverviewImage :on-click="onClick" />
```

Comment:
looks unclear. It'd be better to send events directly in component. Don't pass callbacks through the whole app

Reply:
Try to find or create a module for Nuxt. 

It's expected to see something like


```ts

// utils/analytics/index.ts
enum AnalyticEvent {
  GetStartedClicked = 'Get Started — Clicked`,
  // ...
}
```
```ts
// LandingHeader.vue
<UiButton @click="track" />


function track(): void {
  this.$amplitude.track(AnalyticEvent.GetStartedClicked)
}
```

Reply:
Tried to make it